api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -740,7 +751,249 @@ public void shouldTestPurgeProgramAttributeType() throws Exception {
                 int totalAttributeTypes = pws.getAllProgramAttributeTypes().size();
 		pws.purgeProgramAttributeType(programAttributeType);
 		assertEquals((totalAttributeTypes - 1), pws.getAllProgramAttributeTypes().size());
-}
+	}
+	
+	
+	@Test
+	public void getPrograms_shouldTestGetPrograms() {
+		List<Program> programs = pws.getPrograms("MAL");
+		List<Program> programs1 = pws.getPrograms("PR");
+		assertEquals(programs.size(), 1);
+		assertEquals(programs1.size(), 3);
+	}
+	
+	@Test
+	public void retireProgram_shouldTestRetireProgram() {
+		Program programTest = pws.getAllPrograms().get(0);
+		Program program = pws.retireProgram(programTest, "expired");
+		for (ProgramWorkflow workflow : program.getWorkflows()) {
+			assertEquals(workflow.getRetired(), true);
+			for (ProgramWorkflowState state : workflow.getStates()) {
+				assertEquals(state.getRetired(), true);
+			}
+		}
+		
+	}
+	
+	@Test
+	public void unretireProgram_shouldTestUnretireProgram() {
+		Program program = pws.getAllPrograms().get(0);
+		Date lastModifiedDate = program.getDateChanged();
+		Program program1 = pws.unretireProgram(program);
+		assertEquals(program.getRetired(), false);
+		for (ProgramWorkflow workflow : program.getAllWorkflows()) {
+			if (lastModifiedDate != null && lastModifiedDate.equals(workflow.getDateChanged())) {
+				assertEquals(workflow.getRetired(), false);
+				for (ProgramWorkflowState state : workflow.getStates()) {
+					if (lastModifiedDate.equals(state.getDateChanged())) {
+						assertEquals(state.getRetired(), false);
+					}
+				}
+			}
+		}
+	}
+	
+	@Test
+	public void saveUnretireProgram_shouldTestSaveUnretireProgram() {
+		Program program = new Program();
+		Concept concept = Context.getConceptService().getAllConcepts().get(0);
+		User user = Context.getUserService().getAllUsers().get(0);
+		program.setConcept(concept);
+		program.setDescription("test");
+		program.setName("programTest");
+		program.setCreator(user);
+		List<Program> list = pws.getAllPrograms();
+		String uuid = program.getUuid();
```

Comment:
unnecessary variable. You have only 1 usage of this variable.

Reply:
I removed it

Reply:
I don't see it. What I meant is the suggested change in line 813.